x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/InferenceRequestStats.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.inference;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class InferenceRequestStats implements ToXContentObject, Writeable {
+    private final InferenceFeatureSetUsage.ModelStats modelStats;
+    private final String modelId;
+
+    public InferenceRequestStats(String service, TaskType taskType, @Nullable String modelId, long count) {
+        this(new InferenceFeatureSetUsage.ModelStats(service, taskType, count), modelId);
+    }
+
+    private InferenceRequestStats(InferenceFeatureSetUsage.ModelStats modelStats, @Nullable String modelId) {
+        this.modelStats = new InferenceFeatureSetUsage.ModelStats(modelStats);
+        this.modelId = modelId;
+    }
+
+    public InferenceRequestStats(StreamInput in) throws IOException {
+        this.modelStats = new InferenceFeatureSetUsage.ModelStats(in);
+        this.modelId = in.readOptionalString();
+    }
+
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        builder.field("service", modelStats.service());
+        builder.field("task_type", modelStats.taskType().toString());
```

Comment:
The only reason I chose not to call `modelStats.addXContentFragment()` here is because that causes `task_type` to be all upper case. Maybe we should be consistent and keep with that. Generally we use `TaskType.toString()` to get the value but in `ModelStats` we're using `name()`. I think for most values in the telemetry service things are lowercase. I looked into changing it from `name()` to `toString()` but the telemetry visualizations have hardcoded configurations that explicitly looking for the uppercase values. 🤷‍♂️